packages/web-twig/src/Resources/components/Button/Button.twig

```diff
@@ -28,8 +28,8 @@
     {{ mainProps(props, _allowedAttributes) }}
     {{ classProp(_classNames) }}
     type="{{ _type }}"
-    {{ _disabledAttr | raw }}
-    {{ _onClickAttr | raw }}
+    {{ _disabledAttr  }}
+    {{ _onClickAttr  }}
```

Comment:
Still not sure about escaping the `onClick` prop. By design, the developer is responsible for passing the function that will work. If I escape something for him/her I think I could break the functionality of the script passed 🤔 

Now, I am using `escape('html_attr')` in combination with the `raw` filter. In `"alert('\" test')"` the backslash is escaped and it seems to me not right. Because there should be some escaping using backslashes needed 🤔

Reply:
@literat, I think that everything that developer writes should be executable. He's not hacking his own code (usually).

CMS content is a different thing - there should be an option list provided. Option list that allows/disallows particular attributes to be escaped or (better) not rendered at all. But this is not the problem we're facing here now.

You provide an escaping test in [packages/web-twig/tests/components-fixtures/buttonWithMainProps.twig](https://github.com/lmc-eu/spirit-design-system/pull/742/files#diff-0ba83c2ecd2cdb345eb51e4cf895775dcfb15cc83449042471e0be8509eff831) and you just pass manually escaped quote. Maybe it is a lack in my imagination but I just can't figure out why the escape filter is being used with `html_attr` parameter.

I can't figure out why would I pass HTML attributes directly to `onClick` attribute.

When I pass 

`onClick="document.body.append('<span>hello there</span>'); return false;"`

it appends just a text, not DOM HTML element.

-----------------

Please talk me out of this or keep the door open.

🤔

Reply:
Well, we should discuss this because this is the best solution I can get by now. See snapshots for the Button component. I have tried your example and the inner HTML is always escaped. I have tried the `raw` filter all the way down and nothing works to pass down the HTML.